src/mongocxx/pipeline.hpp

```diff
@@ -174,9 +175,16 @@ class MONGOCXX_API pipeline {
     pipeline& unwind(std::string field_name);
 
     ///
+    /// @return A view of the underlying BSON array this pipeline represents.
+    ///
+    bsoncxx::array::view view_array() const;
+
+    ///
     /// @return A view of the underlying BSON document this pipeline represents.
     ///
-    bsoncxx::document::view view() const;
+    /// @deprecated The view_array() method should be used instead of this method.
+    ///
+    MONGOCXX_DEPRECATED bsoncxx::document::view view() const;
```

Comment:
```
commit a75b38f15319e53f2a611d63613c30eac4f3b63f
Author: J. Rassi <rassi@10gen.com>
Date:   Sat Oct 1 15:49:56 2016 -0400

    CXX-999 Add pipeline::view_array(), deprecate pipeline::view()

    This commit provides a mechanism to get the underlying array out of a
    pipeline object, which logically wraps an array builder.  This is
    needed for database commands which take a pipeline option as a BSON
    array type.

    The pipeline::view() method exposes a document wrapper around this
    array, which is strictly less useful.
```

In other words: pipelines are arrays, and we should expose them as arrays.  Note that an `array::view` can be converted to a `document::view`, but a `document::view` can't be converted to an `array::view`.  It's a design error that `pipeline::view()` didn't return an array originally; hence the addition of the new method and deprecation of the existing one (there's no point in maintaining both).